src/components/ThreeDS2/components/Challenge/Do3DS2Challenge.tsx

```diff
@@ -7,15 +7,20 @@ import getProcessMessageHandler from '../../../../utils/get-process-message-hand
 import { encodeBase64URL } from '../utils';
 import promiseTimeout from '../../../../utils/promiseTimeout';
 import { CHALLENGE_TIMEOUT, UNKNOWN_CHALLENGE_RESOLVE_OBJECT_TIMEOUT, UNKNOWN_CHALLENGE_RESOLVE_OBJECT } from '../../config';
+import { Do3DS2ChallengeProps, Do3DS2ChallengeState } from './types';
 
 const iframeName = 'threeDSIframe';
 
-class Do3DS2Challenge extends Component {
+class Do3DS2Challenge extends Component<Do3DS2ChallengeProps, Do3DS2ChallengeState> {
+    private readonly iframeCallback;
+    private processMessageHandler;
+    private challengePromise: { cancel: () => void; promise: Promise<any> };
+
     constructor(props) {
         super(props);
 
         this.iframeCallback = () => {
-            this.setState({ status: 'iframeLoaded' });
+            this.state = { status: 'iframeLoaded' };
```

Comment:
Is it ok to set state directly?

Reply:
I think this function should be moved out of the constructor

Reply:
Yes, that seems like the best solution here.

Reply:
Mine was also a question for my own understanding. `setState` or `this.state`?

Reply:
The rule of thumb is that you use `this.state` in the constructor and `this.setState` everywhere else.

Reply:
ok, thanks